common/kvstore/src/test/java/org/apache/spark/util/kvstore/RocksDBSuite.java

```diff
@@ -381,6 +382,40 @@ public void testSkipAfterDBClose() throws Exception {
     assertFalse(iter.skip(1));
   }
 
+  @Test
+  public void testResourceCleaner() throws Exception {
+    File dbPathForCleanerTest = File.createTempFile(
+      "test_db_cleaner.", ".rdb");
+    dbPathForCleanerTest.delete();
+
+    RocksDB dbForCleanerTest = new RocksDB(dbPathForCleanerTest);
+    for (int i = 0; i < 8192; i++) {
+      dbForCleanerTest.write(createCustomType1(i));
+    }
+
+    RocksDBIterator<CustomType1> rocksDBIterator = (RocksDBIterator<CustomType1>) dbForCleanerTest.view(CustomType1.class).iterator();
+    Reference<RocksDBIterator<?>> reference = dbForCleanerTest.getRocksDBIteratorRef(rocksDBIterator);
+    assertNotNull(reference);
+    RocksIterator it = rocksDBIterator.internalIterator();
+    // it has not been closed yet, isOwningHandle should be true.
+    assertTrue(it.isOwningHandle());
+    // Manually set rocksDBIterator to null, to be GC.
+    rocksDBIterator = null;
+    // 100 times gc, the rocksDBIterator should be GCed.
+    int count = 0;
+    while (count < 100 && !reference.refersTo(null)) {
+      System.gc();
+      count++;
+      Thread.sleep(100);
+    }
+    // check rocksDBIterator should be GCed
+    assertTrue(reference.refersTo(null));
+    // Verify that the Cleaner will be executed after a period of time, and it.isOwningHandle() will become false.
+    assertTimeout(java.time.Duration.ofSeconds(5), () -> assertFalse(it.isOwningHandle()));
+
+    dbForCleanerTest.close();
```

Comment:
Let's move `dbForCleanerTest.close()` to finally block and t would be better to add `FileUtils.deleteQuietly(dbPathForCleanerTest);` after close in finally block

Reply:
```
dbPathForCleanerTest.delete();
```
can it be removed?

Reply:
OK, it's fine as long as we can ensure there are no residues after testing.

Reply:
sorry, if remove it, there is an exception.
```
org.rocksdb.RocksDBException: `/Users/zhaomin/Files/project/spark/common/kvstore/target/tmp/test_db_cleaner.6329921973463604622.rdb' exists but is not a directory

	at org.rocksdb.RocksDB.open(Native Method)
	at org.rocksdb.RocksDB.open(RocksDB.java:249)
	at org.apache.spark.util.kvstore.RocksDB.<init>(RocksDB.java:122)
	at org.apache.spark.util.kvstore.RocksDB.<init>(RocksDB.java:116)
```